entc/gen/template/tx.tmpl

```diff
@@ -29,25 +29,67 @@ type Tx struct {
 
 	// completion callbacks.
 	mu         sync.Mutex
-	onCommit   []func(error)
-	onRollback []func(error)
+	onCommit   []CommitHook
+	onRollback []RollbackHook
+
+	// ctx lives for the life of the transaction. It is
+	// the same context used by the underlying connection.
+	ctx context.Context
 }
 
-{{- range $func := list "Commit" "Rollback" }}
+{{- range $func, $iface := dict "Commit" "Committer" "Rollback" "Rollbacker" }}
```

Comment:
dict doesn't define order on keys so we might get generated changes without touching this template. Use `keys` for iteration which keep the keys sorted and get the value with `get`.